test/promql/tests/test_controller_reconciliation_errors.yaml

```diff
@@ -0,0 +1,91 @@
+evaluation_interval: 1m
+
+rule_files:
+  - ../extracted-rules.yaml
+
+tests:
+
+  # Alerted test cases
+  - interval: 1m
+    input_series:
+      # Controller with continuous errors greater than 1% so this is alerted
+      - series: 'controller_runtime_reconcile_errors_total{controller="c1-controller", namespace="prod-ns1", pod="pod-1", container="c1", service="s1"}'
+        values: '1+1x19'
+      - series: 'controller_runtime_reconcile_total{controller="c1-controller", namespace="prod-ns1", pod="pod-1", container="c1", service="s1"}'
+        values: '1+1x19'
+
+    alert_rule_test:
+      - eval_time: 20m
+        alertname: ControllerReconciliationErrors
+        exp_alerts:
+          - exp_labels:
+              severity: warning
+              namespace: prod-ns1
+              pod: pod-1
+              controller: c1-controller
+              container: c1
+              service: s1
+            exp_annotations:
+              summary: "Controller c1-controller has reconciliation errors greater than 1%"
+              description: "Controller c1-controller with Pod pod-1 for namespace prod-ns1 has total reconciliation errors greater than 1% compared to total reconciliations in last 15 minutes."
+
+  # Not Alerted test cases
+  - interval: 1m
+    input_series:
+      # Controller with error throughout with value greater than 1% but it's constant series so it's not alerted
+      - series: 'controller_runtime_reconcile_errors_total{controller="c5-controller", namespace="prod-ns5", pod="pod-5", container="c1", service="s1"}'
+        values: '1x19'
+      - series: 'controller_runtime_reconcile_total{controller="c5-controller", namespace="prod-ns5", pod="pod-5", container="c1", service="s1"}'
+        values: '10x19'
```

Comment:
This simulates an idle controller to me. The error total will always be 1 and the reconcile total 10 so there's no activity over those 19m.

The test is useful but the description isn't clear.